libnmstate/nm/route.py

```diff
@@ -0,0 +1,129 @@
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+from operator import itemgetter
+from libnmstate.error import NmstateInternalError
+from libnmstate.schema import Route
+
+from libnmstate import iplib
+
+NM_MAIN_ROUTE_TABLE_ID = 0
+
+
+def get_running(acs_and_ip_cfgs):
+    """
+    Query running routes
+    The acs_and_ip_cfgs should be generate to generate a tuple:
+        NM.NM.ActiveConnection, NM.IPConfig
+    """
+    routes = []
+    for (active_connection, ip_cfg) in acs_and_ip_cfgs:
+        if not ip_cfg.props.routes:
+            continue
+        iface_name = _get_iface_name(active_connection)
+        if not iface_name:
+            raise NmstateInternalError(
```

Comment:
The exception could be raised in `_get_iface_name()` directly. Also it seems to me that it would make more sense to pass the interface name instead of the active connection to get_running() and get_config().

Reply:
If that activeconnection has no routes on it, there is no need to query its interface name or even raise exception for unknown iface name.
But clearly I do more investigation on this. How about we address this in another PR.